ui/app/components/project-config-variables/list-item.ts

```diff
@@ -0,0 +1,80 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+import { ConfigVar, Project } from 'waypoint-pb';
+import { inject as service } from '@ember/service';
+import ApiService from 'waypoint/services/api';
+import FlashMessagesService from 'waypoint/services/flash-messages';
+
+interface VariableArgs {
+  variable: ConfigVar.AsObject;
+  isEditing: boolean;
+  isCreating: boolean;
+  saveVariableSettings: (variable: ConfigVar.AsObject, deleteVariable?: boolean) => Promise<Project.AsObject>;
+  deleteVariable: (variable: ConfigVar.AsObject) => Promise<void>;
+  cancelCreate: () => void;
+}
+
+export default class ProjectConfigVariablesListItemComponent extends Component<VariableArgs> {
+  @service api!: ApiService;
+  @service flashMessages!: FlashMessagesService;
+
+  initialVariable!: ConfigVar.AsObject;
+  @tracked variable: ConfigVar.AsObject;
+  @tracked isCreating: boolean;
+  @tracked isEditing: boolean;
+
+  constructor(owner: unknown, args: VariableArgs) {
+    super(owner, args);
+    let { variable, isEditing, isCreating } = args;
+    this.variable = variable;
+    this.isEditing = isEditing;
+    this.isCreating = isCreating;
+    this.storeInitialVariable();
+  }
+
+  get isEditable(): boolean {
+    // Temporarily making Dynamic vars uneditable
+    return !this.variable.dynamic;
+  }
+
+  storeInitialVariable(): void {
+    this.initialVariable = JSON.parse(JSON.stringify(this.variable));
+  }
+
+  @action
+  async deleteVariable(variable: ConfigVar.AsObject): Promise<void> {
+    await this.args.deleteVariable(variable);
+  }
+
+  @action
+  editVariable(): void {
+    this.isEditing = true;
+    this.storeInitialVariable();
+  }
+
+  @action
+  async saveVariable(e: Event): Promise<void> {
+    e.preventDefault();
+    if (this.variable.name === '' || this.variable.pb_static === '') {
+      return this.flashMessages.error('Variable keys or values can not be empty');
```

Comment:
I’m seeing a TS error here. I think it’s unhappy about the return value. Maybe move the return to the line below?

```
Type 'FlashMessageService' is not assignable to type 'void'.
```

Reply:
Yeah, I was seeing that too. Will update.